tests/filecheck/transforms/acc-dedup.mlir

```diff
@@ -0,0 +1,181 @@
+// CHECK-NEXT:     %init = acc2.setup on "snax_hwpe_mult" ("A" = %A : i32, "B" = %B : i32, "O" = %O : i32, "size" = %c32 : i32) : !acc2.state<"snax_hwpe_mult">
+//                 ^^^ first setup should be untouched ^^^
+// CHECK-NEXT:     %token = "acc2.launch"(%init) <{"param_names" = [], "accelerator" = "snax_hwpe_mult"}> : (!acc2.state<"snax_hwpe_mult">) -> !acc2.token<"snax_hwpe_mult">
+// CHECK-NEXT:     "acc2.await"(%token) : (!acc2.token<"snax_hwpe_mult">) -> ()
+// CHECK-NEXT:     %lb_1 = arith.constant 0 : index
+// CHECK-NEXT:     %ub_1 = arith.constant 100 : index
+// CHECK-NEXT:     %step_1 = arith.constant 1 : index
+// CHECK-NEXT:     %A_shift = arith.addi %A, %c32 : i32
+// CHECK-NEXT:     %20 = acc2.setup on "snax_hwpe_mult" ("A" = %A_shift : i32) in_state(%init) : !acc2.state<"snax_hwpe_mult">
+//                       ^^^^^^^^^^^^^^^^^^^^^ new setup op for loop-invariant variables
+// CHECK-NEXT:     %res_state = scf.for %iv_1 = %lb_1 to %ub_1 step %step_1 iter_args(%inner_state_1 = %20) -> (!acc2.state<"snax_hwpe_mult">) {
+// CHECK-NEXT:       %B_shift = arith.addi %B, %c32 : i32
+// CHECK-NEXT:       %O_shift = arith.addi %O, %c32 : i32
+// CHECK-NEXT:       %s_new_1 = acc2.setup on "snax_hwpe_mult" ("B" = %B_shift : i32, "O" = %O_shift : i32) in_state(%inner_state_1) : !acc2.state<"snax_hwpe_mult">
+//                           only loop-dependent variables left ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// CHECK-NEXT:       %tok = "acc2.launch"(%s_new_1) <{"param_names" = [], "accelerator" = "snax_hwpe_mult"}> : (!acc2.state<"snax_hwpe_mult">) -> !acc2.token<"snax_hwpe_mult">
+// CHECK-NEXT:       "acc2.await"(%tok) : (!acc2.token<"snax_hwpe_mult">) -> ()
+// CHECK-NEXT:       scf.yield %s_new_1 : !acc2.state<"snax_hwpe_mult">
+// CHECK-NEXT:     }
+// CHECK-NEXT:     %final = acc2.setup on "snax_hwpe_mult" ("A" = %A : i32, "B" = %B : i32, "O" = %O : i32) in_state(%res_state) : !acc2.state<"snax_hwpe_mult">
+//                                                size parameter can be inferred as unchanged and deleted ^
+// CHECK-NEXT:     %token2 = "acc2.launch"(%final) <{"param_names" = [], "accelerator" = "snax_hwpe_mult"}> : (!acc2.state<"snax_hwpe_mult">) -> !acc2.token<"snax_hwpe_mult">
+// CHECK-NEXT:     "acc2.await"(%token2) : (!acc2.token<"snax_hwpe_mult">) -> ()
+// CHECK-NEXT:     func.return
+// CHECK-NEXT:   }
+
+
+func.func @nested_loops(%A : i32, %lb : i32, %ub : i32, %step : i32) {
+  %0 = acc2.setup on "simple" () : !acc2.state<"simple">
+  %1 = scf.for %i = %lb to %ub step %step iter_args(%2 = %0) -> (!acc2.state<"simple">) : i32 {
+    %3 = scf.for %j = %i to %ub step %step iter_args(%4 = %2) -> (!acc2.state<"simple">) : i32 {
+      %out_state = acc2.setup on "simple" ("A" = %A : i32, "i" = %i : i32, "j" = %j : i32) in_state(%4) : !acc2.state<"simple">
+      scf.yield %out_state : !acc2.state<"simple">
+    }
+    scf.yield %3 : !acc2.state<"simple">
+  }
+  func.return
+}
+
```

Comment:
There's no CHECK here

Reply:
that way it's never wrong :upside_down_face: